dataactvalidator/filestreaming/loaderUtils.py

```diff
@@ -25,85 +28,62 @@ def compareRecords (recordA,recordB, fields) :
-                        row[field] = Validator.padToLength(row[field],padLength)
-                    if "skip_duplicates" in options:
-                        if row[field] is None or len(row[field].strip()) == 0 or row[field] in valuePresent[field]:
-                            # Value not provided or already exists, skip it
-                            skipInsert = True
-                        else:
-                            # Insert new value
-                            valuePresent[field][row[field]] = True
-                record = model(**row)
-                if not skipInsert:
-                    try:
-                        interface.session.merge(record)
-                        interface.session.commit()
+        # toss out blank rows
+        data.dropna(inplace=True, how='all')
+
+        # Fix up cells that have spaces instead of being empty.
+        # Set the truly empty cells to None so they get inserted to db as NULL
+        # TODO: very ugly function below...is there a better way?
+        data.applymap(lambda x: str(x).strip() if len(str(x).strip()) else None)
+
+        # clean the dataframe column names
+        data.rename(columns=lambda x: cls.cleanColNamesFunction(x), inplace=True)
+        # make sure all values in fieldMap parameter are in the dataframe/csv file
+        for field in fieldMap:
+            if field not in list(data.columns):
+                raise ValueError("{} is required for loading table{}".format(field, model))
+        # toss out any columns from the csv that aren't in the fieldMap parameter
+        data = data[list(fieldMap.keys())]
+        # rename columns as specified in fieldMap
+        data = data.rename(columns=fieldMap)
 
-                    except IntegrityError as e:
-                        # Hit a duplicate value that violates index, skip this one
-                        print("".join(["Warning: Skipping this row: ",str(row)]))
-                        print("".join(["Due to error: ",str(e)]))
-                        interface.session.rollback()
-                        continue
-            interface.session.commit()
+        # apply column options as specified in fieldOptions param
+        for col, options in fieldOptions.items():
+            if "pad_to_length" in options:
+                # pad to specified length
+                data['{}'.format(col)] = data['{}'.format(col)].apply(
+                    lambda x: cls.padFunction(x, options['pad_to_length']))
+            if 'strip_commas' in options and options['strip_commas']:
+                # remove commas for specified column
+                # get rid of commas in dollar amounts
+                data[col] = data[col].str.replace(",", "")
 
+        # add created_at and updated_at columns
+        data = data.assign(
+            created_at=cls.currentTimeFunction, updated_at=cls.currentTimeFunction)
 
+        return data
+
+    @classmethod
+    def insertDataframe(cls, df, table, engine):
```

Comment:
I'm torn too. We had discussed the ramifications of bypassing the ORM shortly before joined us, and agreed that we could do so as part of the domain values load.

That said, this did bite me a few times during development and testing, so I'm amenable to re-thinking this. However, I'm thinking that the benefits of getting this in sooner so we can test the pieces of the load and making sure the deploy works with all this new data outweigh the costs of refactoring at this point.

What do you think?